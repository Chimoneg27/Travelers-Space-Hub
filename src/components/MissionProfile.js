import { useSelector } from 'react-redux';

const MissionProfile = () => {
  const { missions } = useSelector((state) => state.missions);
  const missionsJoin = missions.filter((mission) => mission.mission_join === true);
  return (
    <>
      {

missionsJoin.map((mission) => (
  <div
    key={mission.mission_id}
    className="mission-name-profile"
  >
    <h3>{mission.mission_name}</h3>
  </div>
))
}
    </>
  );
};
export default MissionProfile;
