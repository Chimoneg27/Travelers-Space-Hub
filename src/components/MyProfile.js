import React from 'react';
import Navigation from './Navigation';

const MyProfile = () => (
  <div className="my-profile-page">
    <Navigation />
    <h1>My Profile</h1>
  </div>
);

export default MyProfile;
