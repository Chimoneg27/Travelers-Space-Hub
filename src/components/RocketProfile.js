import { useSelector } from 'react-redux';

import React from 'react';

const RocketProfile = () => {
  const rockets = useSelector((state) => state.rockets.rocketArr);
  const rocketsJoin = rockets.filter((rocket) => rocket.reserved === true);

  return (
    <div className="Rocket-Profile">
      {
        rocketsJoin.map((rocket) => (
          <div
            key={rocket.id}
            className="rocket-name-profile"
          >
            <h3>{rocket.name}</h3>
          </div>
        ))
      }
    </div>
  );
};

export default RocketProfile;
