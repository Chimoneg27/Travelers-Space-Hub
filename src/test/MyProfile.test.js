import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { configureMockStore } from '@reduxjs/toolkit';
import MissionProfile from '../components/MissionProfile';
import RocketProfile from '../components/RocketProfile';
import MyProfile from '../components/MyProfile';

const mockStore = configureMockStore([]);

describe('MyProfile', () => {
  test('renders MyProfile component correctly', async () => {
    const store = mockStore({
      rocket: {
        rocketArr: [
          {
            id: '1',
            name: 'Rocket1',
          },
          {
            id: '2',
            name: 'Rocket2',
          },
        ],
        isDataFetched: true,
      },
      mission: {
        missionArr: [
          {
            id: '1',
            name: 'Mission1',
          },
          {
            id: '2',
            name: 'Mission2',
          },
        ],
        isLoading: true,
      },
    });
    const ProfileComponent = () => (
      <Provider store={store}>
        <MyProfile>
          <MissionProfile />
          <RocketProfile />
        </MyProfile>
      </Provider>
    );
    const { getText } = render(<ProfileComponent />);
    expect(getText).toMatchSnapshot();
    expect(screen.getByText('My Missions')).toBeInTheDocument();
    expect(screen.getByText('My Rockets')).toBeInTheDocument();
  });
});
