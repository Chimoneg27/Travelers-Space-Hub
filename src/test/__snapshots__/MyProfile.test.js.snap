// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test My Profile page Check if the component has changed 1`] = `
<div
  className="my-profile-page"
>
  <div
    className="missions-sec"
  >
    <h2>
      My Missions
    </h2>
  </div>
  <div
    className="rockets-sec"
  >
    <h2>
      My Rockets
    </h2>
    <div
      className="Rocket-Profile"
    />
  </div>
</div>
`;
